runtime/vm/resolvefield.cpp

```diff
@@ -578,6 +583,124 @@ addHiddenInstanceField(J9JavaVM *vm, const char *className, const char *fieldNam
 	return 0;
 }
 
+/**
+ * Report a hot field if the JIT has determined that the field has met appropriate thresholds to be determined a hot field. 
+ * Valid if dynamicBreadthFirstScanOrdering is enabled.
+ *
+ * @param javaVM[in] pointer to the J9JavaVM
+ * @param reducedCpuUtil normalized cpu utilization of the method reporting the hot field
+ * @param clazz pointer to the class where a hot field should be added/updated
+ * @param fieldOffset value of the field offset that should be added/updated as a hot field for the given class
+ * @param reducedFrequency normalized block frequency of the hot field for the method reporting the hot field
+ */
+void
+reportHotField(J9JavaVM *javaVM, int32_t reducedCpuUtil, J9Class* clazz, U_8 fieldOffset,  U_32 reducedFrequency)
+{
+	/* exit if pools are null */
+	if (NULL == clazz->classLoader->hotFieldPool || NULL == javaVM->hotFieldClassInfoPool) {
+		return;
+	}
+
+	/* If hotFieldsInfo pool element for the class does not exist already, create and initialize the class' hotFieldsInfo pool element
+	 * otherwise, create/update the given hot field for the class 
+	 */
+	if (NULL == clazz->hotFieldsInfo) {
+		createClassHotFieldsInfo(javaVM, clazz, fieldOffset, reducedCpuUtil, reducedFrequency);
+	} else if (clazz->hotFieldsInfo->hotFieldListLength < MAX_HOT_FIELD_LIST_LENGTH) {
+		addOrUpdateHotField(clazz, fieldOffset, reducedCpuUtil, reducedFrequency);
+	}
+}
+
+/**
+ * Create and initialize a class' hotFieldsInfo pool element.
+ * Valid if dynamicBreadthFirstScanOrdering is enabled.
+ *
+ * @param javaVM[in] pointer to the J9JavaVM
+ * @param clazz pointer to the class where a hot field should be added
+ * @param fieldOffset value of the field offset that should be added as a hot field for the given class
+ * @param reducedCpuUtil normalized cpu utilization of the method reporting the hot field
+ * @param reducedFrequency normalized block frequency of the hot field for the method reporting the hot field
+ */
+VMINLINE void
```

Comment:
you  should return a success/failure return code, if this fails do not attempt `addOrUpdateHotField`

Reply:
Based on our discussion I will mark this as resolved as we decided that we will not set any error/success return code.